### DISCUSSION TITLE
Authentification

### DISCUSSION THREAD

**Discussion by Hootlook** [NuGet/Home#11848] [General]
Hello, 
We are trying to Nuget packages through a custom client using this api, the remote feed is on DevOps and protected thus when trying to download a package i get a 401 error in code. 

How do you setup the PackageManager to handle authentification in code (preferably through windows identification) ?

PS: I tried with a nuget.config file with some credentials inside loaded in code in through "new PackageSourceProvider(settings)" but it doesn't look like its picking up on it...

**Comment by Hootlook** [comment]
Fix here => https://github.com/NuGet/Home/issues/7714

**Comment by erdembayar** [comment]
We recently updated docs here, please check examples: https://docs.microsoft.com/en-us/azure/devops/artifacts/nuget/dotnet-exe?view=azure-devops
